pyoverkiz/servers/overkiz_server.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from json import JSONDecodeError
+from typing import Any
+
+from aiohttp import ClientResponse, ClientSession
+from attr import define
+
+from pyoverkiz.exceptions import (
+    AccessDeniedToGatewayException,
+    BadCredentialsException,
+    InvalidCommandException,
+    InvalidEventListenerIdException,
+    InvalidTokenException,
+    MaintenanceException,
+    MissingAuthorizationTokenException,
+    NoRegisteredEventListenerException,
+    NotAuthenticatedException,
+    NotSuchTokenException,
+    SessionAndBearerInSameRequestException,
+    TooManyAttemptsBannedException,
+    TooManyConcurrentRequestsException,
+    TooManyExecutionsException,
+    TooManyRequestsException,
+    UnknownObjectException,
+    UnknownUserException,
+)
+from pyoverkiz.types import JSON
+
+
+@define(kw_only=True)
+class OverkizServer(ABC):
+    """Class to describe an Overkiz server."""
+
+    name: str
+    endpoint: str
+    manufacturer: str
+    session: ClientSession
+
+    configuration_url: str | None
+
+    @abstractmethod
+    async def login(self, username: str, password: str) -> bool:
+        """Login to the server."""
```

Comment:
```suggestion
        """Login to the server."""
        pass
```
Isn't this the writing style for 'interfaces' in Python?

Reply:
Body can be empty, the doctring is enough for readability.